submit.go

```diff
@@ -438,7 +438,11 @@ func saveLogPart(logNum int, filename string, reader io.Reader, reportDir string
 	// Either way, we need to append .gz, because we're compressing it.
 	var leafName string
 	if logRegexp.MatchString(filename) {
-		leafName = filename + ".gz"
+		if strings.HasSuffix(filename, ".gz") {
+			leafName = filename
+		} else {
+			leafName = filename + ".gz"
+		}
```

Comment:
won't we still compress the log file even if it's already gzipped? maybe that's a silly thing to do, but it should be called `foo.log.gz.gz` if we're doing it.

Reply:
We unzip the file as part of the `compressed-log` form field handling, then in this code we gzip it again. I think the only place this would go wrong is if someone uploaded a gzip file to the uncompressed `log` form field (already an issue, you're meant to upload to compressed-log) - if you name it with a gz then you end up with only one gz, not two; if you don't name it with a gz you also get one.

The code would also do the wrong thing, if you upload a file for instance "foo.gz" in the `log` form-field - since the filename doesn't match the expression, it's overwritten with the fixed "logs-0000.log.gz` filename - so that too would get the compression status out of sync with the filename.

I looked at doing something in the `compressed-log` side before it came in to this method, but trying to add something into gzip.NewPart handling to update a Header if it happens to be called 'filename' and ends 'gz' seemed more work.

Reply:
> We unzip the file as part of the compressed-log form field handling, then in this code we gzip it again

right, fair enough. In that case, could I ask you to add a comment to explain that this is what's going in?

Reply:
I've updated the comment here.